modules/javafx.graphics/src/main/java/com/sun/javafx/util/Utils.java

```diff
@@ -872,7 +872,7 @@ public static void forceInit(final Class<?> classToInit) {
 
     public static boolean assertionEnabled() {
         boolean assertsEnabled = false;
-        assert assertsEnabled = true;  // Intentional side-effect !!!
+        assert (assertsEnabled = true) == true;
```

Comment:
Is it worth leaving the "Intentional side-effect" comment?

Reply:
i think *now* the code is clear without the comment!
(and the comment does not explain *why* it is intentional or what is the intention)

Reply:
I would probably leave it as before. I found that code already when working on the OpenJFX project and while it looks strange at first glance, it was clear afterwards. And the comment makes this clear as well.

The `== true` might confuse more now.

Reply:
The combination of removing the comment in connection with adding the `== true` has made the purpose of this line of code somewhat less clear. It needs a comment to explain what is otherwise a pretty obscure construct. The code relies on the fact that the body of the `assert` is executed if-and-only-if assertions are enabled at runtime. Thus the side-effect of the assignment is crucial to its operation. Here is a comment that might explain this:

```
            // The following assertion check will always pass. The side-effect
            // of the `assertsEnabled = true` assignment is intentional. It will
            // be executed if-and-only-if assertions are enabled at runtime.
```